test/TestINode.java

```diff
@@ -0,0 +1,38 @@
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
+public class TestINode {
+	
+	WordsEnvironment environment = new WordsEnvironment();
+
+	/*********************
+	 * EvalPosition 
+	 ********************/
+	AST numLeaf1 = new LNode(AST.ASTType.NUM, 0, 0.0);
+	AST numLeaf2 = new LNode(AST.ASTType.NUM, 0, 2.1);
+	AST stringLeafCoercable = new LNode(AST.ASTType.STRING, 0, "2.1");
```

Comment:
It seems like you're suggesting we would pre-create a lot of "ready-to-use" AST nodes that can be used in different combinations in each test case.  But since the coercion method modifies the AST node itself, won't the coercion happen just one time (by whichever test is run first), and later tests that use this node will not really be testing whether they have incorporated coercion correctly?